app/ui/browser/webview-controller.js

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import assert from 'assert';
+import { EventEmitter } from 'events';
+import { isUUID } from './browser-util';
+
+const getPageIndexById = (pages, id) => pages.findIndex(p => p.id === id);
+
+/**
+ * A utility class that extends EventEmitter to receive page IDs and
```

Comment:
Comment on how this fits into the larger architecture.  I expected one controller per webview, but in fact there's a single controller for all the webviews.